plugins/languages/treesitter/treesitter.nix

```diff
@@ -6,210 +6,169 @@
-                scope_incremental = cfg.incrementalSelection.keymaps.scopeIncremental;
-                node_decremental = cfg.incrementalSelection.keymaps.nodeDecremental;
-              };
-            }
-          else
-            null;
-
-        indent = if cfg.indent then { enable = true; } else null;
-
-        ensure_installed = if cfg.nixGrammars then [ ] else cfg.ensureInstalled;
-        ignore_install = cfg.ignoreInstall;
-        parser_install_dir = cfg.parserInstallDir;
-      } // cfg.moduleConfig;
-    in
-    mkIf cfg.enable {
-      extraConfigLua =
-        (optionalString (cfg.parserInstallDir != null) ''
-          vim.opt.runtimepath:append("${cfg.parserInstallDir}")
-        '')
-        + ''
-          require('nvim-treesitter.configs').setup(${helpers.toLuaObject tsOptions})
-        ''
-        + (optionalString (cfg.languageRegister != { }) ''
-          __parserFiletypeMappings = ${helpers.toLuaObject cfg.languageRegister}
-
-          for parser_name, ft in pairs(__parserFiletypeMappings) do
-            require('vim.treesitter.language').register(parser_name, ft)
-          end
-        '');
-
-      extraFiles = mkIf cfg.nixvimInjections {
-        "queries/nix/injections.scm" = ''
-          ;; extends
-
-          (binding
-            attrpath: (attrpath (identifier) @_path)
-            expression: [
-              (string_expression (string_fragment) @lua)
-              (indented_string_expression (string_fragment) @lua)
-            ]
-            (#match? @_path "^extraConfigLua(Pre|Post)?$"))
-
-          (binding
-            attrpath: (attrpath (identifier) @_path)
-            expression: [
-              (string_expression (string_fragment) @vim)
-              (indented_string_expression (string_fragment) @vim)
-            ]
-            (#match? @_path "^extraConfigVim(Pre|Post)?$"))
-        '';
-      };
-
-      extraPlugins =
-        if cfg.nixGrammars then [ (cfg.package.withPlugins (_: cfg.grammarPackages)) ] else [ cfg.package ];
-      extraPackages = with pkgs; [
-        tree-sitter
-        nodejs
-        cfg.gccPackage
-      ];
-
-      opts = mkIf cfg.folding {
-        foldmethod = "expr";
-        foldexpr = "nvim_treesitter#foldexpr()";
-      };
+    moduleConfig = helpers.defaultNullOpts.mkAttrsOf types.anything { } ''
+      This is the configuration for extra modules. It should not be used directly.
+    '';
+
+    nixGrammars = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Install grammars with Nix";
     };
+
+    nixvimInjections = mkEnableOption "nixvim specific injections, like lua highlighting in extraConfigLua";
+  };
+
+  extraConfig = cfg: {
+    extraConfigLua =
+      (optionalString (cfg.parserInstallDir != null) ''
+        vim.opt.runtimepath:append("${cfg.parserInstallDir}")
+      '')
+      + ''
+        require('nvim-treesitter.configs').setup(${helpers.toLuaObject cfg.settings // cfg.moduleConfig})
```

Comment:
This feels wrong (`settings // moduleConfig`).

It'd be better if whatever your other options need to change in the `setup` opts did so by configuring the nix settings options, e.g. with mkDefault, or mkForce.

This allows end-users to override the definitions.

Can't tell from my initial phone review what `moduleConfig` is for though, so can't offer more specific advice atm.

Reply:
Yeah, I was just commenting on this on that other thread, too... I agree it feels weird after looking back at the derivation.  It seemingly was used to support other modules modifying treesitter from an internal option, which shouldn't be needed with the new settings attribute set.

Reply:
Do you think having a rename alias `moduleConfig -> settings` would work as a graceful deprecation then?

Other than the old option being nullable the types should match up, I think.

Reply:
Makes sense, renaming instead of just removing it. I was curious how many plugins actually tied into the moduleConfig, but hadn't scoured the repo, yet.

Reply:
Was thinking in case end-users were using the option, but if it was an internal option I think it's fair to use a `mkRemovedOptionModule` assertion or even just remove it with no warnings